src/lib/GoogleSpreadsheet.ts

```diff
@@ -108,7 +118,8 @@ export class GoogleSpreadsheet {
     /** id of google spreadsheet doc */
     spreadsheetId: SpreadsheetId,
     /** authentication to use with Google Sheets API */
-    auth: GoogleApiAuth
+    auth: GoogleApiAuth,
+    rateLimitedRetryConfig?: RateLimitedRetryConfig
```

Comment:
Since we dont know what other options may be needed in the future, this should probably just be a more generic `options` param with these options nested. We also probably should provide a simple opt-in with sensible defaults. Perhaps something like
```
options?: {
  retryOnRateLimit?: true | RateLimitedRetryConfig
}
```

Reply:
Sounds good, I'll implement something similar to the suggestion in the API documentation so, seems best to follow their best practices.